d2/src/test/java/com/linkedin/d2/jmx/D2ClientJmxManagerTest.java

```diff
@@ -17,165 +17,440 @@
 package com.linkedin.d2.jmx;
 
 import com.linkedin.d2.balancer.LoadBalancerStateItem;
+import com.linkedin.d2.balancer.dualread.DualReadModeProvider;
+import com.linkedin.d2.balancer.dualread.DualReadStateManager;
 import com.linkedin.d2.balancer.properties.ClusterProperties;
 import com.linkedin.d2.balancer.properties.ServiceProperties;
+import com.linkedin.d2.balancer.properties.UriProperties;
 import com.linkedin.d2.balancer.simple.ClusterInfoItem;
+import com.linkedin.d2.balancer.simple.SimpleLoadBalancer;
 import com.linkedin.d2.balancer.simple.SimpleLoadBalancerState;
+import com.linkedin.d2.balancer.strategies.relative.RelativeLoadBalancerStrategy;
 import com.linkedin.d2.balancer.util.canary.CanaryDistributionProvider;
 import com.linkedin.d2.balancer.util.partitions.PartitionAccessor;
+import com.linkedin.d2.discovery.stores.file.FileStore;
 import java.net.URI;
 import java.util.Collections;
+import java.util.concurrent.ScheduledExecutorService;
 import java.util.concurrent.atomic.AtomicLong;
 import org.mockito.Captor;
 import org.mockito.ArgumentCaptor;
 import org.mockito.Mock;
 import org.mockito.Mockito;
 import org.mockito.MockitoAnnotations;
 import org.testng.Assert;
-import org.testng.annotations.BeforeMethod;
+import org.testng.annotations.DataProvider;
 import org.testng.annotations.Test;
 
 import static org.mockito.Mockito.*;
 
 
 public class D2ClientJmxManagerTest {
-  @Mock
-  SimpleLoadBalancerState _simpleLoadBalancerState;
-  @Mock
-  JmxManager _jmxManager;
```

Comment:
I've put these mocks into a fixture because direct mock member vars inside the class can't be used in a data provider. For example, when I create different D2ClientJmxManager with different params but use the same _jmxManager mock:

`@DataProvider
public Object[][] dataProvider()
{
   return new Object[][]
   {
      {new D2ClientJmxManager(..., _jmxManager)}
      {new D2ClientJmxManager(..., _jmxManager, ...ZK, null)}
      {new D2ClientJmxManager(..., _jmxManager, ....xDS, null)}
   };
}`
Using this data provider in a test method (such as `testSetSimpleLBStateListenerUpdateClusterInfo`), the test will fail with errors saying the jmxManager is null.

Reply:
Seems fine by me, it's easier to pass in one object than 10 !